agent/src/test/java/com/newrelic/agent/android/AndroidAgentImplTest.java

```diff
@@ -306,19 +309,20 @@ public void start() {
         Assert.assertNotNull("Should contain app background event", getSessionEvent(queuedEvents));
 
         agentImpl.applicationForegrounded(e);
-        assertEquals("Should contain foreground user action events", eventManager.getEventsRecorded(), 1);
+        assertEquals("Should contain foreground user action events", eventManager.getEventsRecorded(), 4);
```

Comment:
Why would the counts here (and below) be higher? What are the additional events?

Reply:
Because when we persist the events, the total counts are growing while going foreground/background without sending payload: 
When DT is enabled, it changed from 1,3,1,3 --> 1,3,4,6
When DT is not enabled, it changed from 0,1,0,1 --> 0,1,1,2

But if we would like to simulate the similar scenario while app goes to background will send one last payload, I can clean the eventStore.

Reply:
Yes, the test should reflect the actual lifecycle and state of the agent. With DT enabled, it shows the test is not resetting the event manager state when the agent restarts.